pyro/distributions/empirical.py

```diff
@@ -0,0 +1,160 @@
+from __future__ import absolute_import, division, print_function
+
+import numbers
+
+import math
+import torch
+
+from pyro.distributions.torch_distribution import TorchDistribution
+from pyro.distributions.torch import Categorical
+from pyro.distributions.util import copy_docs_from, is_validation_enabled
+from pyro.distributions.util import log_sum_exp
+
+
+@copy_docs_from(TorchDistribution)
+class Empirical(TorchDistribution):
+    r"""
+    Empirical distribution associated with the sampled data.
+    """
+
+    arg_constraints = {}
```

Comment:
Is there a sensible `.support`? Does it make most sense to set `support = constraints.real`?

Reply:
Yeah, I suppose so.

Reply:
Are `enumerate_support` and `support` expected to satisfy any invariants in `TorchDistributions`?

Reply:
Some informal invariants are:
- `.enumerate_support()` should return a set of samples whose probabilities sum to 1
- arguments to `.log_prob()` should return `-inf` if outside of `.support`
- return values of `.sample()` and `.enumerate_support()` shoule lie in the `.support`

I don't know whether this is important; it just surprised me to see `.arg_constraints` without `.support`.